File: packages/react-native/ReactCommon/runtimeexecutor/ReactCommon/RuntimeExecutor.cpp
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#include "RuntimeExecutor.h"

namespace facebook::react {

std::mutex& getMainThreadMutex() {
  static std::mutex mainThreadMutex;
  return mainThreadMutex;
}

} // namespace facebook::react

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/RuntimeExecutor.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge

import com.facebook.jni.HybridClassBase
import com.facebook.proguard.annotations.DoNotStrip

/** A Kotlin holder for a C++ RuntimeExecutor. */
@DoNotStrip public class RuntimeExecutor @DoNotStrip private constructor() : HybridClassBase()
